ddl/partition.go

```diff
@@ -884,6 +899,16 @@ func onDropTablePartition(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 			return ver, errors.Trace(err)
 		}
 		physicalTableIDs = removePartitionInfo(tblInfo, partNames)
+	}
+
+	rules := buildPlacementDropRules(job.SchemaID, tblInfo.ID, physicalTableIDs)
+
+	err = infosync.UpdatePlacementRules(nil, rules)
+	if err != nil {
+		return ver, errors.Wrapf(err, "failed to notify PD the placement rules")
+	}
```

Comment:
If it fails due to `getGlobalInfoSyncer` or pd unavailable, the placement rules on PD is untouched, so I think it can be cancelled. It will be meaningless to retry forever.

Reply:
Is it reallt acceptable to cancel a drop action? I will fix it later. I am outside.

Reply:
If an error occurs before requesting the PD, the error is not evitable by retrying, so it have to be cancelled.

Reply:
Got it. I though PD is partially unavailable. But that is rare case, right?